chapters/connectors.tex

```diff
@@ -786,26 +786,28 @@ \section{Equation Operators for Overconstrained Connection-Based Equation System
 
 \subsection{Overconstrained Equation Operators for Connection Graphs}\label{overconstrained-equation-operators-for-connection-graphs}
 
-A type or record declaration may have an optional definition of function
-\lstinline!equalityConstraint! that shall have the following prototype:
+A type or record declaration may have an optional definition of function \lstinline!equalityConstraint! that shall have the following prototype:
+% TODO: Can't have angle brackets and \emph in the same mathescape due to LaTeXML issue:
+% - https://github.com/brucemiller/LaTeXML/issues/1477
+% Once the issue has been fixed, change to single mathescape for better character spacing.
 \begin{lstlisting}[language=modelica]
 type Type // overdetermined type
-  extends <base type>;
+  extends $\langle$$\mbox{\emph{base type}}$$\rangle$;
   function equalityConstraint // non-redundant equality
     input Type T1;
     input Type T2;
-    output Real residue[ <n> ];
+    output Real residue[$n$];
```

Comment:
I'm not sure about skipping angles here (and the similar line below). The n here should normally be replaced by a number (such as 3 for quaternions).

Reply:
Couldn't we solve this nicely by rephrasing this slightly and placing it before the listing?
> The \lstinline!residue! output of the \lstinline!equalityConstraint! function shall have known size, say constant $n$.

Reply:
I also like having it as `<n>`.

Reply:
I don't find the angle bracket construct very pretty in general, but still useful for placeholders in need of a textual description.  For something acting as a placeholder for a number, I find it more consistent to use the rather well established use of "meta variables" typeset as math.

Reply:
What are the well-established meta-variables? Where are they established?

Reply:
The main example is in the descriptions of built-in functions.  Since we have a lot of functions that are best explained using mathematical notation, it really helps typesetting to use math style meta variables in the listings that illustrate how a function is used.  Besides making the mathematical descriptions look much better in the text, it also helps separating the fixed names (function identifiers, named arguments, etc) from the "variable" parts.

You can find plenty of examples here: https://specification.modelica.org/master/arrays.html#built-in-array-functions

Reply:
I feel like the cases in the built-in functions are different from this one.  The built-in function definitions are given out of context so you do not need to know how exactly A and n are defined. The definitions in the connector section appear as complete definitions and parts in <> signify that the exact definition in this case is irrelevant.

Maybe, for consistency with the rest of the definition, we should replace "[<n>]" with "[<integer constant>]" or whatever the actual type is supposed to be? What does "known" mean exactly? Can it be a parameter that can be evaluated at translation time? Or can it be enumeration or Boolean?

Reply:
(I can see what you meant by looking at the underlying markdown, but you might want to edit your comment so that the quoted brackets are displayed as code.)

If the thing in angle brackets would have been _Integer constant_, I agree that this would be a case for angle brackets.  However, we already have _n_ in the text, and I don't see why we wouldn't want it to be connected with the listing by using it in there as well?

Reply:
> (I can see what you meant by looking at the underlying markdown, but you might want to edit your comment so that the quoted brackets are displayed as code.)

Ugh, one time I didn't check the preview before posting.... Edited now.

Reply:
Yeah, and I should probably stop extending my comments so much after first posting, as these additions don't appear in email notifications.

Reply:
Ehmm.. I'm a bit lost here, what is the outcome?

Reply:
I think we're still actively discussing what to do, but if I can't convince the two of you that the already present use of _n_ in the surrounding text is reason enough to also use _n_ in the listing, I'll of course have to give it up and go for `[<Integer constant>]` so that this can be merged.

Reply:
It strikes me now that the use of _n_ in the listing would actually assume that it is an `Integer` dimension – is this really the current intention?  To me, the current text would actually allow an array indexed by `Boolean` or enumeration (both of which will result in a constant size, which is the requirement), and I don't see that it would really be a problem to do so.  If we want to make it more clear that this is allowed, I suggest this:
```suggestion
    output Real residue[$\langle$$\mbox{\emph{constant size dimension}}$$\rangle$];
```
In the text, we could elaborate this further:
> The \lstinline!residue! output of the \lstinline!equalityConstraint! function shall have known size, say constant $n$.  (That is, $n$ cannot depend on inputs to the function, and the array cannot have flexible size given by colon (\lstinline!:!).)

Reply:
I would be OK with removing the angled brackets around n. But I feel like we need top open a separate issue to clarify what "known size"  means. If we come up with a concise wording we could consider putting that into the angle brackets in the code.

Reply:
Created #2886.

Reply:
Good.  @HansOlsson, would you then be OK with the resolution to move (with some rephrasing) the sentence introducing _n_ to before the listing?

Reply:
I would be ok with different variants of that - and I agree with @eshmoylova that it seems best to have a separate issue for that, so resolving, accepting and merging this.